src/server/cluster/incoming_slot_migration.h

```diff
@@ -46,6 +47,16 @@ class IncomingSlotMigration {
     return source_id_;
   }
 
+  void ReportError(dfly::GenericError err) {
+    std::lock_guard lk(error_mu_);
+    last_error_ = err;
+  }
+
+  std::string GetErrorStr() const {
+    std::lock_guard lk(error_mu_);
+    return last_error_.Format();
+  }
+
```

Comment:
not sure why this is duplicated when the context stores an error by itself

Reply:
if the error into the context the process will be canceled,